pkg/coveragedb/coveragedb.go

```diff
@@ -81,23 +99,26 @@ func SaveMergeResult(ctx context.Context, client spannerclient.SpannerClient, de
 	ssCache := make(map[string][]string)
 
 	session := uuid.New().String()
-	mutations := []*spanner.Mutation{}
+	var mutations []*spanner.Mutation
 
 	for {
-		var mcr MergedCoverageRecord
-		err := dec.Decode(&mcr)
+		var wr JSONLWrapper
+		err := dec.Decode(&wr)
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
 			return rowsCreated, fmt.Errorf("dec.Decode(MergedCoverageRecord): %w", err)
 		}
-		if mcr.FileData == nil {
-			return rowsCreated, errors.New("field MergedCoverageRecord.FileData can't be nil")
+		if mcr := wr.MCR; mcr != nil {
+			mutations = append(mutations, fileRecordMutation(session, mcr))
+			subsystems := getFileSubsystems(mcr.FilePath, ssMatcher, ssCache)
+			mutations = append(mutations, fileSubsystemsMutation(descr.Namespace, mcr.FilePath, subsystems))
+		} else if fl := wr.FL; fl != nil {
+			mutations = append(mutations, fileFunctionsMutation(session, fl))
+		} else {
+			return rowsCreated, errors.New("JSONLWrapper can't be empty")
 		}
-		mutations = append(mutations, fileRecordMutation(session, &mcr))
-		subsystems := fileSubsystems(mcr.FilePath, ssMatcher, ssCache)
-		mutations = append(mutations, fileSubsystemsMutation(descr.Namespace, mcr.FilePath, subsystems))
 		// There is a limit on the number of mutations per transaction (80k) imposed by the DB.
 		// This includes both explicit mutations of the fields (6 fields * 1k records = 6k mutations)
```

Comment:
Should this comment be updated?

Reply:
Thanks, no. The number of fields in the 'functions' table is <6.